### PR TITLE
Resolved issue #35 and added new backend stuff

### LINKED ISSUE #35: Integrate mongodb calls to CourseInfoProvider servlet
Moving forward with the fact that we have all the courses in the db now, let's rewrite the servlet `CourseInfoProvider.java` so that it grabs the data from the db instead of that random csv file (`courseDB.csv`).

### PR DESCRIPTION
**Pull request by Davidster**
resolves #35 
## Resolved issue #35 

The servlet `CourseInfoProvider.java` now uses our MongoDB DB in order to retrieve the course info. I had to make **very** slight changes to js code to accomodate this change cuz the format isn't equivalent.

### In the backend:

I've added a new serlvet named `FilterCourseCodes`, which responds to `POST` requests at the url path `filtercoursecodes` (see `web.xml`). 
This new servlet expects a string in the request body named `filter`. It then searches through our `courseData` DB and returns the first 25 course codes which contain said string, as a JSON array.

Here is an example of the servlet in operation (request body is in higher box, response is in lower box):

image

### In the frontend:

We had already implemented auto-complete in our search box. I made the necessary changes to make the same autocomplete work with this new backend servlet. 
This change was surprisingly easy to do cuz [jquery](http://api.jqueryui.com/autocomplete/) 😍.

You can go ahead and try it out at http://138.197.6.26/courseplannerd/sequenceBuilder.html

### ...[victory](https://www.youtube.com/watch?v=oL7N7PiPUVs)

### CODE REVIEW

File: src/main/java/CourseInfoProvider.java [hunk 1]
```diff
@@ -22,60 +22,29 @@ public void doPost(HttpServletRequest request,
...
+            responseString = dbResult.toJson();
+        } else {
+            responseString = "{}";
         }
 
+        logger.info("Responding with: " + responseString);
+        PrintWriter out = response.getWriter();
+        out.println(responseString);
     }
 }
```
Comment (PeterGhimself) [inline src/main/java/CourseInfoProvider.java]: Looks much cleaner with the mongo

File: src/main/webapp/js/script.js [hunk 2]
```diff
@@ -342,43 +342,34 @@ function fillCourseInfoBox(courseInfo){
...
-            if(winterIncluded){
-                termsOffered = termsOffered + "winter";
-                if(summerIncluded)
-                    termsOffered += ", ";
-            }
-            if(summerIncluded){
-                termsOffered = termsOffered + "summer";
-            }
-        }
+        // commenting this out as we are not supporting this feature for our MVP
```
Comment (PeterGhimself) [inline src/main/webapp/js/script.js]: If we're not using this code, we night as well leave it out of master. It will still be saved in a commit.

Comment (Davidster) [inline src/main/webapp/js/script.js]: but then we have to mewmorize a commmit ID

Comment (PeterGhimself) [inline src/main/webapp/js/script.js]: We can put it in the wiki for reference as a hyperlink

Comment (Davidster) [inline src/main/webapp/js/script.js]: sounds like a bad idea to me

**Comment by Davidster** [comment]
@PeterGhimself I fixed the commented out code little luluboobmoom please accept this peace offering and approve my changes <3333333

**Pull request merged**